.esbuild/server.ts

```diff
@@ -0,0 +1,96 @@
+import express from 'express';
+import type { NextFunction, Request, Response } from 'express';
+import cors from 'cors';
+import { getBuildConfig } from './util.js';
+import { context } from 'esbuild';
+import chokidar from 'chokidar';
+
+const mermaidCtx = await context(
+  getBuildConfig({ minify: false, core: false, entryName: 'mermaid' })
+);
+const mermaidIIFECtx = await context(
+  getBuildConfig({ minify: false, core: false, entryName: 'mermaid', format: 'iife' })
+);
+const externalCtx = await context(
+  getBuildConfig({ minify: false, core: false, entryName: 'mermaid-example-diagram' })
+);
+const zenumlCtx = await context(
+  getBuildConfig({ minify: false, core: false, entryName: 'mermaid-zenuml' })
+);
+const contexts = [mermaidCtx, mermaidIIFECtx, externalCtx, zenumlCtx];
+
+const rebuildAll = async () => {
+  const time = Date.now();
+  await Promise.all(contexts.map((ctx) => ctx.rebuild()));
+  console.log('Rebuilt in' + (Date.now() - time) + 'ms');
```

Comment:
This is the most nit-picky of nit-picks, since it doesn't really matter here, but `Date.now()` shouldn't be used for calculating stuff like this.

JavaScript has a [`performance.now()`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now) function instead (see the MDN docs on why `Date.now()` is bad).

Buuuut, we could even use [`console.time()` and `console.timeEnd()`](https://developer.mozilla.org/en-US/docs/Web/API/console/time) instead:

```suggestion
  console.time("Rebuilt");
  await Promise.all(contexts.map((ctx) => ctx.rebuild()));
  console.timeEnd("Rebuilt");
```

This will print something that looks like `Rebuilt: 2763.864013671875 ms`.

The only issue is that using `console.time()`/`console.timeEnd()` won't work if `rebuildAll()` is being called multiple times at the same time (is that an issue?).

Reply:
Haha, yeah. Changed :)